In [8]:
import re

def extract_log_section(file_path):
    # Regular expression pattern to match the required section
    pattern = re.compile(
        r"(>>>>>>>testing : .+?mse:\d+\.\d+, mae:\d+\.\d+, dtw:-999)",
        re.DOTALL
    )
    
    extracted_sections = []

    with open(file_path, 'r') as file:
        content = file.read()

        # Find all matches in the content
        matches = pattern.findall(content)
        extracted_sections.extend(matches)

    return extracted_sections

def main():
    # Path to the log file
    log_file_path = "scripts/long_term_forecast/Exchange_script/"

    # Extract the sections
    extracted_sections = extract_log_section(log_file_path)

    # Print the extracted sections
    for section in extracted_sections:
        print(section)
        print("=" * 50)  # Separator for clarity

if __name__ == "__main__":
    main()


In [2]:
import re

def extract_log_section(file_path):
    # Regular expression pattern to match the required section
    pattern = re.compile(
        r"(>>>>>>>testing : .+?mse:\d+\.\d+, mae:\d+\.\d+, dtw:-999)",
        re.DOTALL
    )
    
    extracted_sections = []

    with open(file_path, 'r') as file:
        content = file.read()

        # Find all matches in the content
        matches = pattern.findall(content)
        extracted_sections.extend(matches)

    return extracted_sections

def main():
    # Path to the log file
    log_file_path = "scripts/long_term_forecast/Exchange_script/iTransformer.log"

    # Extract the sections
    extracted_sections = extract_log_section(log_file_path)

    # Print the extracted sections
    for section in extracted_sections:
        print(section)
        print("=" * 50)  # Separator for clarity

if __name__ == "__main__":
    main()


>>>>>>>testing : long_term_forecast_Exchange_96_96_iTransformer_custom_ftM_sl96_ll48_pl96_dm64_nh8_el2_dl1_df64_expand2_dc4_fc3_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1422
test shape: (1422, 96, 8) (1422, 96, 8)
test shape: (1422, 96, 8) (1422, 96, 8)
mse:0.08852455765008926, mae:0.2094336897134781, dtw:-999
>>>>>>>testing : long_term_forecast_Exchange_96_192_iTransformer_custom_ftM_sl96_ll48_pl192_dm64_nh8_el2_dl1_df64_expand2_dc4_fc3_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1326
test shape: (1326, 192, 8) (1326, 192, 8)
test shape: (1326, 192, 8) (1326, 192, 8)
mse:0.18049836158752441, mae:0.3027716279029846, dtw:-999
>>>>>>>testing : long_term_forecast_Exchange_96_336_iTransformer_custom_ftM_sl96_ll48_pl336_dm32_nh8_el2_dl1_df32_expand2_dc4_fc3_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1182
test shape: (1182, 336, 8) (1182, 336, 8)
test shape: (1182, 336, 8) (1182, 336, 8)
mse:0.3627845048904419, mae:0.44005173444747925, dtw:-99

In [3]:
import re

def extract_log_section(file_path):
    # Regular expression pattern to match the required section
    pattern = re.compile(
        r"(>>>>>>>testing : .+?mse:\d+\.\d+, mae:\d+\.\d+, dtw:-999)",
        re.DOTALL
    )
    
    extracted_sections = []

    with open(file_path, 'r') as file:
        content = file.read()

        # Find all matches in the content
        matches = pattern.findall(content)
        extracted_sections.extend(matches)

    return extracted_sections

def main():
    # Path to the log file
    log_file_path = "scripts/long_term_forecast/Exchange_script/TimesNet.log"

    # Extract the sections
    extracted_sections = extract_log_section(log_file_path)

    # Print the extracted sections
    for section in extracted_sections:
        print(section)
        print("=" * 50)  # Separator for clarity

if __name__ == "__main__":
    main()


>>>>>>>testing : long_term_forecast_Exchange_96_96_TimesNet_custom_ftM_sl96_ll48_pl96_dm64_nh8_el2_dl1_df64_expand2_dc4_fc3_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1422
test shape: (1422, 96, 8) (1422, 96, 8)
test shape: (1422, 96, 8) (1422, 96, 8)
mse:0.10529237240552902, mae:0.23513366281986237, dtw:-999
>>>>>>>testing : long_term_forecast_Exchange_96_192_TimesNet_custom_ftM_sl96_ll48_pl192_dm64_nh8_el2_dl1_df64_expand2_dc4_fc3_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1326
test shape: (1326, 192, 8) (1326, 192, 8)
test shape: (1326, 192, 8) (1326, 192, 8)
mse:0.23448333144187927, mae:0.35271111130714417, dtw:-999
>>>>>>>testing : long_term_forecast_Exchange_96_336_TimesNet_custom_ftM_sl96_ll48_pl336_dm32_nh8_el2_dl1_df32_expand2_dc4_fc3_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1182
test shape: (1182, 336, 8) (1182, 336, 8)
test shape: (1182, 336, 8) (1182, 336, 8)
mse:0.3632594645023346, mae:0.43862369656562805, dtw:-999
>>>>>>>t

### MSE Results Table

| pred_len | iTransformer | TimesNet | Mamba |
|----------|--------------|----------|-------|
| **96**   | 0.0885       | 0.1053   | 0.1274|
| **192**  | 0.1805       | 0.2345   | 0.2763|
| **336**  | 0.3628       | 0.3633   | 0.7388|
| **720**  | 0.9077       | 0.9352   | 3.2842|


### MAE Results Table

| pred_len | iTransformer | TimesNet | Mamba |
|----------|--------------|----------|-------|
| **96**   | 0.2094       | 0.2351   | 0.2580|
| **192**  | 0.3028       | 0.3527   | 0.3854|
| **336**  | 0.4401       | 0.4386   | 0.6398|
| **720**  | 0.7243       | 0.7367   | 1.2959|


Seems like iTransformer is SOTA on time series forecasting, according to the metrics and datasets given.

Prediction on S & P 500 Index next month return:

Should we focus on mean or Variance of return? Variance -> Vol -> Vol related derivatives.

In [25]:
import re

def extract_log_sections(file_path):
    # Regular expression pattern to match the required section
    pattern = re.compile(
        r"(>>>>>>>testing : .+?mse:\d+\.\d+, mae:\d+\.\d+, dtw:-999, r2:\d+\.\d+)",
        re.DOTALL
    )
    
    extracted_sections = []

    with open(file_path, 'r') as file:
        content = file.read()

        # Find all matches in the content
        matches = pattern.findall(content)
        extracted_sections.extend(matches)

    return extracted_sections

def main():
    # Path to the log file
    log_file_path = "scripts/long_term_forecast/Exchange_script/Mamba.log"

    # Extract the sections
    extracted_sections = extract_log_sections(log_file_path)

    # Print the extracted sections
    for section in extracted_sections:
        print(section)
        print("=" * 50)  # Separator for clarity

if __name__ == "__main__":
    main()


>>>>>>>testing : long_term_forecast_Exchange_96_96_Mamba_custom_ftM_sl96_ll48_pl96_dm128_nh8_el2_dl1_df16_expand2_dc4_fc1_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1422
test shape: (1422, 96, 8) (1422, 96, 8)
test shape: (1422, 96, 8) (1422, 96, 8)
mse:0.12744960188865662, mae:0.2580082416534424, dtw:-999, r2:0.9263931512832642
>>>>>>>testing : long_term_forecast_Exchange_192_192_Mamba_custom_ftM_sl192_ll48_pl192_dm128_nh8_el2_dl1_df16_expand2_dc4_fc1_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1326
test shape: (1326, 192, 8) (1326, 192, 8)
test shape: (1326, 192, 8) (1326, 192, 8)
mse:0.2763102948665619, mae:0.38538235425949097, dtw:-999, r2:0.8364956378936768
>>>>>>>testing : long_term_forecast_Exchange_336_336_Mamba_custom_ftM_sl336_ll48_pl336_dm128_nh8_el2_dl1_df16_expand2_dc4_fc1_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1182
test shape: (1182, 336, 8) (1182, 336, 8)
test shape: (1182, 336, 8) (1182, 336, 8)
mse:0.7387800216674805, 

In [10]:
import re

def extract_log_sections(file_path):
    # Regular expression pattern to match the required section
    pattern = re.compile(
        r"(>>>>>>>testing : .+?mse:\d+\.\d+, mae:\d+\.\d+, dtw:-999, r2:\d+\.\d+)",
        re.DOTALL
    )
    
    extracted_sections = []

    with open(file_path, 'r') as file:
        content = file.read()

        # Find all matches in the content
        matches = pattern.findall(content)
        extracted_sections.extend(matches)

    return extracted_sections

def main():
    # Path to the log file
    log_file_path = "scripts/long_term_forecast/Exchange_script/TimesNet.log"

    # Extract the sections
    extracted_sections = extract_log_sections(log_file_path)

    # Print the extracted sections
    for section in extracted_sections:
        print(section)
        print("=" * 50)  # Separator for clarity

if __name__ == "__main__":
    main()


>>>>>>>testing : long_term_forecast_Exchange_96_96_TimesNet_custom_ftM_sl96_ll48_pl96_dm64_nh8_el2_dl1_df64_expand2_dc4_fc3_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1422
test shape: (1422, 96, 8) (1422, 96, 8)
test shape: (1422, 96, 8) (1422, 96, 8)
mse:0.10529204457998276, mae:0.2351333498954773, dtw:-999, r2:0.9391899704933167
>>>>>>>testing : long_term_forecast_Exchange_96_192_TimesNet_custom_ftM_sl96_ll48_pl192_dm64_nh8_el2_dl1_df64_expand2_dc4_fc3_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1326
test shape: (1326, 192, 8) (1326, 192, 8)
test shape: (1326, 192, 8) (1326, 192, 8)
mse:0.23420313000679016, mae:0.3516334593296051, dtw:-999, r2:0.8614122271537781
>>>>>>>testing : long_term_forecast_Exchange_96_336_TimesNet_custom_ftM_sl96_ll48_pl336_dm32_nh8_el2_dl1_df32_expand2_dc4_fc3_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1182
test shape: (1182, 336, 8) (1182, 336, 8)
test shape: (1182, 336, 8) (1182, 336, 8)
mse:0.3632518649101257

In [14]:
import re

def extract_log_sections(file_path):
    # Regular expression pattern to match the required section
    pattern = re.compile(
        r"(>>>>>>>testing : .+?mse:\d+\.\d+, mae:\d+\.\d+, dtw:-999, r2:\d+\.\d+)",
        re.DOTALL
    )
    
    extracted_sections = []

    with open(file_path, 'r') as file:
        content = file.read()

        # Find all matches in the content
        matches = pattern.findall(content)
        extracted_sections.extend(matches)

    return extracted_sections

def main():
    # Path to the log file
    log_file_path = "scripts/long_term_forecast/Exchange_script/iTransformer.log"

    # Extract the sections
    extracted_sections = extract_log_sections(log_file_path)

    # Print the extracted sections
    for section in extracted_sections:
        print(section)
        print("=" * 50)  # Separator for clarity

if __name__ == "__main__":
    main()


>>>>>>>testing : long_term_forecast_Exchange_96_96_iTransformer_custom_ftM_sl96_ll48_pl96_dm64_nh8_el2_dl1_df64_expand2_dc4_fc3_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1422
test shape: (1422, 96, 8) (1422, 96, 8)
test shape: (1422, 96, 8) (1422, 96, 8)
mse:0.08852455765008926, mae:0.2094336897134781, dtw:-999, r2:0.9488738179206848
>>>>>>>testing : long_term_forecast_Exchange_96_192_iTransformer_custom_ftM_sl96_ll48_pl192_dm64_nh8_el2_dl1_df64_expand2_dc4_fc3_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1326
test shape: (1326, 192, 8) (1326, 192, 8)
test shape: (1326, 192, 8) (1326, 192, 8)
mse:0.18049836158752441, mae:0.3027716279029846, dtw:-999, r2:0.8931915760040283
>>>>>>>testing : long_term_forecast_Exchange_96_336_iTransformer_custom_ftM_sl96_ll48_pl336_dm32_nh8_el2_dl1_df32_expand2_dc4_fc3_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 1182
test shape: (1182, 336, 8) (1182, 336, 8)
test shape: (1182, 336, 8) (1182, 336, 8)
mse:0.3627